src/components/SideNav.jsx

```diff
@@ -1,20 +1,48 @@
 import { NavLink } from "react-router-dom";
+import { useDBdata } from "../context/db-data-context";
 import { useMessageHandling } from "../context/message-handling-context";
 import "../stylesheets/sidenav.css";
 import { sidenavItemList } from "../utilities/sidenav-items";
+import { v4 as uuid } from "uuid";
+
 export function SideNav() {
+  const { labels } = useDBdata();
   const { showSidenav, setShowSidenav } = useMessageHandling();
   return (
     <aside
       className={`sidenav-bg ${!showSidenav ? "sidenav-hide" : ""}`}
       onClick={() => setShowSidenav(false)}
     >
       <ul className="list-group-stacked sidenav">
-        {sidenavItemList.map((item) => (
+        <NavLink
+          end={true}
+          to="/notes"
+          key={uuid()}
+          className={({ isActive }) => (isActive ? "link active" : "link")}
+        >
+          <li className="list-item list-w-icon px-2">
+            <i className="material-icons-outlined list-icon">description</i>
+            Notes
+          </li>
+        </NavLink>
+        {labels?.map((item) => (
+          <NavLink
+            end={true}
+            to={`/notes/label/${item}`}
+            key={uuid()}
+            className={({ isActive }) => (isActive ? "link active" : "link")}
+          >
+            <li className="list-item list-w-icon px-2">
+              <i className="material-icons-outlined list-icon">label</i>
+              {item}
+            </li>
+          </NavLink>
+        ))}
+        {sidenavItemList?.map((item) => (
           <NavLink
             end={true}
             to={item.route}
-            key={item.title}
+            key={uuid()}
```

Comment:
Do not use UUID in key as it will always generate new id and key should be unique but should be same for same elements on each render, this will create unnecessary re-renders as always new key will be there

Reply:
Fixed.
I've now changed it's value to `"index-itemTitle"`( something like `1-Notes`, `2-LabelName` and so on..)